### PR TITLE
added tasmanian-magma example

### PR DESCRIPTION
**Pull request by mkstoyanov**
* added tasmanian-magma example

TODO:
- [x] Implement (in CMake) testing for the new example
- [x] Document additions in the existing or new `README.md` files including the main `README.md`

### CODE REVIEW

**Comment by mkstoyanov** [comment]
I guess someone else should take a look at this and approve if OK.

File: tasmanian/CMakeLists.txt
Comment (balos1) [inline tasmanian/CMakeLists.txt:6]: Please use the [xsdk_add_test macro](https://github.com/xsdk-project/xsdk-examples/blob/master/cmake/XsdkAddTest.cmake).

Comment (balos1) [reply tasmanian/CMakeLists.txt:6]: Example: https://github.com/xsdk-project/xsdk-examples/blob/master/sundials/CMakeLists.txt#L41

File: tasmanian/CMakeLists.txt
Comment (balos1) [inline tasmanian/CMakeLists.txt:1]: Move this to the top level CMakeLists file like the other packages (and guard with `ENABLE_TASMANIAN`).

**Review by balos1 — CHANGES_REQUESTED** [review]

**Review by stomov — APPROVED** [review]

File: tasmanian/CMakeLists.txt
Comment (mkstoyanov) [reply tasmanian/CMakeLists.txt:3]: It will compile in any case and it will run just fine with only BLAS or CUDA. Magma is the best option, but it is optional. And even if nothing is enabled, a meaningful message will be given.

The purpose of the example is not to have production code, but documentation in the form of code. So this is good, thanks for checking on it.

File: tasmanian/CMakeLists.txt
Comment (mkstoyanov) [reply tasmanian/CMakeLists.txt:1]: Limited development access at the moment, will update when I get a chance. Thanks for commenting.

**Comment by v-dobrev** [comment]
Merged in branch [testing-v0.4.0](https://github.com/xsdk-project/xsdk-examples/tree/testing-v0.4.0) for testing ...

**Comment by v-dobrev** [comment]
ping: @mkstoyanov, can you please address the @balos1's requests?

**Review by ulrikeyang — APPROVED** [review]
This looks fine except that Cody's comments should be addressed. Thank you.

File: tasmanian/CMakeLists.txt
Comment (mkstoyanov) [reply tasmanian/CMakeLists.txt]: Done!

File: tasmanian/CMakeLists.txt
Comment (balos1) [reply tasmanian/CMakeLists.txt:4]: @mkstoyanov I think this is the last thing that needs to be addresed.

Comment (mkstoyanov) [reply tasmanian/CMakeLists.txt]: Should be good now.

**Comment by v-dobrev** [comment]
Re-merged in branch [testing-v0.4.0](https://github.com/xsdk-project/xsdk-examples/tree/testing-v0.4.0) for testing ...

**Review by balos1 — APPROVED** [review]

**Pull request merged**
